src/backend/recorders/baseRecorder.ts

```diff
@@ -131,6 +134,22 @@ export abstract class BaseRecorder<T extends WebGLObject> implements IRecorder {
     public appendRecordedInformation(capture: ICapture): void {
         capture.frameMemory[this.objectName] = this.frameMemory;
         capture.memory[this.objectName] = this.memoryPerSecond;
+        capture.objects[this.objectName] = {};
+        for (const id in this.objects) {
+            if (this.objects.hasOwnProperty(id)) {
+                const object = this.objects[id].deref();
```

Comment:
The idea of a `WeakRef` was to allow garbage collection on WebGL objects (although I've never seen it in practice yet).

However, objects which no longer exist must also have their `WeakRef` removed from `this.objects`, or those `WeakRef` themselves still leak memory.
We should probably auto-removed dead references from the `this.objects` from time to time.
There could be a garbage collector every N object creations, which would look if any `WeakRef` are dead, to remove the `WeakRef` itself from the list.
`FinalizerRegistry` might work to get auto notification when the list would have to be cleaned up.

There are also `WeakMap` and `WeakSet` but we probably can't use them here, because we can't iterate over them.

Thoughts?